qutebrowser/browser/webkit/webpage.py

```diff
@@ -80,6 +80,10 @@ def __init__(self, win_id, tab_id, tabdata, private, parent=None):
         self.downloadRequested.connect(self.on_download_requested)
         self.unsupportedContent.connect(self.on_unsupported_content)
         self.loadStarted.connect(self.on_load_started)
+
+        self.mainFrame().javaScriptWindowObjectCleared.connect(
+            self.handle_clear_focus)
+
```

Comment:
Can you try moving this to `webkittab.py` instead? I'm trying to slowly move code out of the QWeb(Engine)View subclasses where possible.